### ISSUE TITLE
Some Questions And Bug Reports

### ISSUE THREAD

**Issue by TobishiroGinko** [henrygd/beszel#314]
Hello,

Thank you very much for your hard work on Beszel! I really appreciate the simplicity, elegance, and practicality of this tool.

I noticed that someone has already made related suggestions, among which I am particularly concerned about ICMP Ping Support and a Public Dashboard. Could you please let me know if these features are planned for future releases?

Additionally, I have encountered an issue when logging into the Dashboard using Microsoft Edge. The browser's saved credentials autofill feature causes a login failure.

In the web console, I observed the following errors as shown in the attached screenshots:

![2e769ad9-1393-4b01-be9a-201c23a7dafa](https://github.com/user-attachments/assets/70227085-6a17-4272-8ce0-2b69b00c78b2)

![f52f088e-49a8-4766-9e9d-0beae5301829](https://github.com/user-attachments/assets/d5e59937-b2ae-4c27-85cd-1cdc86a35e40)

**Comment by TobishiroGinko** [comment]
Additionally, I noticed that when viewing the password after entering it, there is no distinction between uppercase and lowercase letters.

**Comment by henrygd** [comment]
Thanks for bringing this to my attention. There is a hidden honeypot field that your browser is filling out.

It's there to offer some basic protection against bots, but I didn't consider this situation. For now please type or copy/paste your email and password.

I will try adding `autoComplete="off"` to the field. Or I may remove it entirely because the next release will add support for rate limiting login attempts.

The lowercase is just CSS and shouldn't affect functionality, but I will remove it because there's no need for it.

---

**ICMP ping** - Yes, I would like to add this to the agent so you can ping services that might not be accessible to the hub. I don't have a hard timeline but possibly in the next few months.

**Public dashboard** - What is the functionality you want for with this ideally? A settings page where admins are able to create public dashboard URLs and add systems to them? Then anyone with the link can view those systems without logging in?

**Comment by AlphonsoKuvalis** [comment]
@henrygd The features he requested are likely from a famous Chinese-made monitoring software called Nezha.

Recently Nezha received a major update and freaks out some of its end users. Most of them started looking for alternatives and Beszel turns out to be a good choice. And, as you can see, they started creating issues to request Beszel to replicate the features of Nezha.

IMHO Nezha and Beszel are fundamentally different. Since one of the most important features of Beszel is its lightweight design, filling it with the features from other software is not worthwhile. It’s like expecting Beszel to record metrics data like Prometheus.

**Comment by henrygd** [comment]
@AlphonsoKuvalis Don't worry, we're on the same page. I haven't lost sight of the original goal of the project, and I won't add a feature if it means sacrificing efficiency or simplicity.

**Comment by stqfdyr** [comment]
@AlphonsoKuvalis 

Yes, you are mostly right

But there is one difference. The reason why we are fed up with Nezha is because Nezha adds more and more bloated and bad functions to a server monitoring, such as: intranet penetration, DDNS, scheduled tasks, webSSH, etc. The agent has high authority. It's terrible. These are not things that a monitor should do. It deviates from the original intention and core of the design, which is simplicity and efficiency.

So we won't let beszel becomes the next Nezha. We choose beszel because it is simple and efficient, right?

**Comment by TobishiroGinko** [comment]
@henrygd  I'm really sorry for the late reply. Just as you said, this is exactly what I needed.The solution I have in mind is to allow users to view the dashboard without logging in, but sensitive information like IPs would be hidden. This way, only logged-in users can see all the details.

**Comment by TobishiroGinko** [comment]
@AlphonsoKuvalis I agree with your point that Beszel should remain lightweight. I am not a user of Nezha; previously, I used [Artemis](https://github.com/kwaitsing/Artemis) for server monitoring, which offered no additional features beyond basic monitoring. However, Artemis was far from perfect and could even be considered too minimalistic. Therefore, I chose Beszel because it is more comprehensive and better aligns with my aesthetic preferences (to me, aesthetics might be more important than functionality).

The functionality I desire does not involve asking Beszel to replicate the features of Nezha. Monitoring ICMP ping data falls within the standard practices of server monitoring. The public dashboard feature would simply allow me to share my server status with others when needed, without having to create a read-only account for each person who needs access to my monitoring data and notify them individually. I do not wish for Beszel to become an all-in-one solution like Nezha. I love Beszel because it is simple, efficient, and elegant. My requests are based solely on what I need, not on replicating "features of other software" (otherwise, why wouldn't I just use that other software instead?).

**Comment by HarleySalas** [comment]
I love the simplicity and the *performance*. 
I understand keeping things simple and avoiding bloat, but wonder what your guys' thoughts are for more types of monitoring, suchas for websites themselves?

I recently looked at [checkmate](https://bluewavelabs.gitbook.io/checkmate) and thought that it's a huge upgrade compared to something like UptimeKuma, but at the same time, it's not the same level as Beszel. I know I'd really love to just use one great piece of software, suchas Beszel and forget about the rest. Are those sort of features things that you may end up considering for Beszel, or are you going to keep it more focused on hardware monitoring alone?

**Comment by henrygd** [comment]
@HarleySalas We'll add basic ping / http service monitoring. It likely won't have as many options as Uptime Kuma, but should cover most use cases.

One advantage that it would provide over Uptime Kuma or Checkmate is that you'll be able to monitor internal services on any system, because we'll use the agents to do the monitoring.

**Comment by henrygd** [comment]
@TobishiroGinko `autoComplete="off"` has been added to the honeypot field in 0.9.0. No rush, but please check if you still have the same problem with the login form after you update.
